src/color_calibration/match_histograms.py

```diff
@@ -1,50 +1,92 @@
 import numpy as np
 import cv2
 
-def create_color_mapping(ref, tgt):
-    color_map = np.zeros(256)
+def create_color_mapping(ref_hist, tgt_hist):
+    """
+    Create a mapping function that maps target intensity values to reference values,
+    based on their normalized cumulative histograms.
+    
+    Args:
+        ref_hist (np.ndarray): Normalized cumulative histogram (CDF) of the reference color card.
+        tgt_hist (np.ndarray): Normalized cumulative histogram (CDF) of the target color card.
+        
+    Returns:
+        np.ndarray: A lookup table mapping each intensity (0-255) from target to reference.
+    """
+    color_map = np.zeros(256, dtype=np.uint8)
     k = 0
-    for src_i in range(len(tgt)):
-        for ref_i in range(len(ref)):
-            if ref[ref_i] >= tgt[src_i]:
-                k = ref_i
-                break
+    for src_i in range(256):
+        while k < 255 and ref_hist[k] < tgt_hist[src_i]:
+            k += 1
         color_map[src_i] = k
     return color_map
 
 def match_histograms(ref_card, tgt_card, tgt_image):
     """
+    Perform channel-wise histogram matching based on the normalized foreground intensities.
+    No resizing is performed—the normalized cumulative histogram is computed on the raw,
```

Comment:
```suggestion
    No resizing is performed; the normalized cumulative histogram is computed on the raw,
```
Rendering with the em-dash is a bit ambiguous and a semicolon works here too.